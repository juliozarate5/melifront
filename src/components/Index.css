body{
    overflow-x: hidden;
    background-color: #e7e7e7;
}


.img-banner{
    width: 100%;
}
.imagen{
    width: 100%;
    height: 250px;
    border-top-right-radius: 15px;
    border-top-left-radius: 15px;
}

.navbar{  
    background-color: #fff159 !important;
}

.navbar-light .navbar-brand{
    font-weight: 600 !important;
    color: #070D6B !important;
}

a{
    cursor: pointer;
}

.custom-card{
    border-radius: 15px;
}

.card:hover{
    opacity: 0.85;
    transition: opacity .6s;
    -moz-transition: opacity .6s;
    -webkit-transition: opacity .6s;
    -o-transition: opacity .6s;

}