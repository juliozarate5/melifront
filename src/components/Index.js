import React, { useEffect, useState } from 'react'
import { getProducts } from '../services/ProductService';
import Container from './products/Container';
import './Index.css';
import Banner from './ui/Banner';
import NavBar from './ui/NavBar';

const title = "Mercado Libre";

export default function Index() {

    const [products, setProducts] = useState([]);
    const [pagination, setPagination] = useState({
        pageNumber: 0,
        pageSize: 5,
        totalPages: 1,
        totalElements: 10,
        first: true,
        last: true,
    });

    const [arrayTotal, setArrayTotal] = useState(new Array(pagination.totalPages).fill(0));

    useEffect(() => {
        const getAll = async () => {
          const {data, pageNumber, pageSize, totalPages, totalElements, first, last} 
          = await getProducts(pagination.pageNumber, pagination.pageSize);
          setProducts(data);
          setPagination({
            ...pagination,
            pageNumber,
            pageSize,
            totalPages,
            totalElements,
            first,
            last
          });
          setArrayTotal(new Array(totalPages).fill(0));
        }
        getAll();
    }, [pagination.pageNumber]);

    const handlePage = (page) => {
        setPagination({
            ...pagination,
            pageNumber: page
        });
    };

    const handleNext = () => {
        setPagination({
            ...pagination,
            pageNumber: pagination.pageNumber + 1
        });
    };

    const handlePrevious = () => {
        setPagination({
            ...pagination,
            pageNumber: pagination.pageNumber - 1
        });
    };

    return (
        <>
            <NavBar title={title} />
            <Banner />
            <Container 
                products={products}
                arrayTotal={arrayTotal}
                pagination={pagination}
                handlePage={handlePage}
                handleNext={handleNext}
                handlePrevious={handlePrevious}
            />
        </>
    )
}
