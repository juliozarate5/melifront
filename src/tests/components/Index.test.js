import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import Index from '../../components/Index';

describe('Index tests', () => {

    test('showing component successful', () => {
        const wrapper = shallow(<Index />)
        expect(wrapper).toMatchSnapshot();
    });

});