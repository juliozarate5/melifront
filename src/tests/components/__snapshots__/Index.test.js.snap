// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index tests showing component successful 1`] = `
<Fragment>
  <NavBar
    title="Mercado Libre"
  />
  <Banner />
  <Container
    arrayTotal={
      Array [
        0,
      ]
    }
    handleNext={[Function]}
    handlePage={[Function]}
    handlePrevious={[Function]}
    pagination={
      Object {
        "first": true,
        "last": true,
        "pageNumber": 0,
        "pageSize": 5,
        "totalElements": 10,
        "totalPages": 1,
      }
    }
    products={Array []}
  />
</Fragment>
`;
